### PR TITLE
Fix unit tests for Windows

### LINKED ISSUE #122: Add windows support
When I try to run `php-scoper add-prefix` in my simple project I get an Exception:

```
[Symfony\Component\Filesystem\Exception\IOException]                          
Failed to create "C:\www\test\public\wp-content\plugins\asdasdasdsad\buildC:\www\test\public\wp-content\plugins\asdasdasdsad": mkdir(): No such file or directory.   
```

As you can see the path is duplicated. Using output dir / working dir parameters doesn't help.

I'm on Windows 10 using the MINGW64 command line.

### PR DESCRIPTION
**Pull request by villfa**
Fix #122 .

I still have 3 errors but I don't know if it's related to Windows.

The details of the errors:
```
1) Humbug\PhpScoper\Console\Command\AddPrefixCommandIntegrationTest::test_scope_in_normal_mode
PHPUnit\Framework\Exception: PHP Fatal error:  Uncaught Exception: Serialization of 'Closure' is not allowed in C:\Users\villfa\AppData\Local\Temp\PHP3EBD.tmp:254
Stack trace:
#0 C:\Users\villfa\AppData\Local\Temp\PHP3EBD.tmp(254): serialize(Array)
#1 C:\Users\villfa\AppData\Local\Temp\PHP3EBD.tmp(730): __phpunit_run_isolated_test()
#2 {main}
  thrown in C:\Users\villfa\AppData\Local\Temp\PHP3EBD.tmp on line 254
Fatal error: Uncaught Exception: Serialization of 'Closure' is not allowed in C:\Users\villfa\AppData\Local\Temp\PHP3EBD.tmp:254
Stack trace:
#0 C:\Users\villfa\AppData\Local\Temp\PHP3EBD.tmp(254): serialize(Array)
#1 C:\Users\villfa\AppData\Local\Temp\PHP3EBD.tmp(730): __phpunit_run_isolated_test()
#2 {main}
  thrown in C:\Users\villfa\AppData\Local\Temp\PHP3EBD.tmp on line 254

2) Humbug\PhpScoper\Console\Command\AddPrefixCommandIntegrationTest::test_scope_in_verbose_mode
PHPUnit\Framework\Exception: PHP Fatal error:  Uncaught Exception: Serialization of 'Closure' is not allowed in C:\Users\villfa\AppData\Local\Temp\PHP4006.tmp:254
Stack trace:
#0 C:\Users\villfa\AppData\Local\Temp\PHP4006.tmp(254): serialize(Array)
#1 C:\Users\villfa\AppData\Local\Temp\PHP4006.tmp(730): __phpunit_run_isolated_test()
#2 {main}
  thrown in C:\Users\villfa\AppData\Local\Temp\PHP4006.tmp on line 254
Fatal error: Uncaught Exception: Serialization of 'Closure' is not allowed in C:\Users\villfa\AppData\Local\Temp\PHP4006.tmp:254
Stack trace:
#0 C:\Users\villfa\AppData\Local\Temp\PHP4006.tmp(254): serialize(Array)
#1 C:\Users\villfa\AppData\Local\Temp\PHP4006.tmp(730): __phpunit_run_isolated_test()
#2 {main}
  thrown in C:\Users\villfa\AppData\Local\Temp\PHP4006.tmp on line 254

3) Humbug\PhpScoper\Console\Command\AddPrefixCommandIntegrationTest::test_scope_in_very_verbose_mode
PHPUnit\Framework\Exception: PHP Fatal error:  Uncaught Exception: Serialization of 'Closure' is not allowed in C:\Users\villfa\AppData\Local\Temp\PHP4121.tmp:254
Stack trace:
#0 C:\Users\villfa\AppData\Local\Temp\PHP4121.tmp(254): serialize(Array)
#1 C:\Users\villfa\AppData\Local\Temp\PHP4121.tmp(730): __phpunit_run_isolated_test()
#2 {main}
  thrown in C:\Users\villfa\AppData\Local\Temp\PHP4121.tmp on line 254
Fatal error: Uncaught Exception: Serialization of 'Closure' is not allowed in C:\Users\villfa\AppData\Local\Temp\PHP4121.tmp:254
Stack trace:
#0 C:\Users\villfa\AppData\Local\Temp\PHP4121.tmp(254): serialize(Array)
#1 C:\Users\villfa\AppData\Local\Temp\PHP4121.tmp(730): __phpunit_run_isolated_test()
#2 {main}
  thrown in C:\Users\villfa\AppData\Local\Temp\PHP4121.tmp on line 254
```

### CODE REVIEW

**Comment by theofidry** [comment]
Thanks for looking into this!

In my experience, those errors happen when an error happen whilst having the `@runInSeparateProcess`. You might need to remove it and execute those tests one by one to see what the real underlying error is

**Comment by villfa** [comment]
The last errors are now fixed.

**Comment by villfa** [comment]
@theofidry would you have time to review this PR?

File: tests/Console/Command/AddPrefixCommandIntegrationTest.php [hunk 1]
```diff
@@ -283,6 +287,35 @@ private function getNormalizeDisplay(string $display): string
         return implode("\n", $lines);
     }
 
+    private function getNormalizeSeparators(string $display): string
```
Comment (theofidry) [inline tests/Console/Command/AddPrefixCommandIntegrationTest.php]: I think this code could be extracted in a re-usable piece. So I would do like in [humbug/box](https://github.com/humbug/box) and create a [`DisplayNormalizer `](https://github.com/humbug/box/blob/master/tests/Console/DisplayNormalizer.php) helper class. I would then also rename `getNormalizeSeparators()` for `normalizeSeparators()`.

The `if ('\\' === DIRECTORY_SEPARATOR) {` check can also be moved here. This way you always call the `normalizeSeparators` method without caring about the platform

File: tests/Console/Command/AddPrefixCommandIntegrationTest.php [hunk 2]
```diff
@@ -283,6 +287,35 @@ private function getNormalizeDisplay(string $display): string
...
+                $fixedPath = str_replace('\\', '/', $path);
+                $display = str_replace($path, $fixedPath, $display);
+            }
+        }
+
+        return $display;
+    }
+
+    private function getNormalizeProgressBar(string $display): string
+    {
```
Comment (theofidry) [inline tests/Console/Command/AddPrefixCommandIntegrationTest.php]: Same comment as for `getNormalizeSeparators()`

**Comment by theofidry** [comment]
>PHPUnit\Framework\Exception: PHP Fatal error:  Uncaught Exception: Serialization of 'Closure' is not allowed in C:\Users\villfa\AppData\Local\Temp\PHP3EBD.tmp:254

That is an error I can help with I think. It tends to happen when the test is annotated with `@ runTestsInSeparateProcesses`. When that happens, the test is run in a separate process and upon failure, the stack trace is returned. I think there is a serialization happening at some point there but since there is a closure in the stack trace it fails with the mentioned error.

The long term plan would be to fix it directly in PHPUnit really. But meanwhile I just temporary disable the `@runTestsInSeparateProcesses` and run each failing test individually.

**Comment by villfa** [comment]
The error in Travis seems unrelated to my changes

**Comment by theofidry** [comment]
Indeed, I've restarted the job

**Comment by theofidry** [comment]
Indeed master seems to be failing, it would be best to fix it before

**Comment by villfa** [comment]
@theofidry My last 2 commits fix the error.
The branch is now stable.

**Comment by theofidry** [comment]
Awesome, thanks very much for the hard work!

**Pull request merged**